core/sdk/src/main/kotlin/com/viam/sdk/core/component/input/InputControllerRPCService.kt

```diff
@@ -0,0 +1,132 @@
+            if (cancelledTriggers.isNotEmpty()) controller.registerControlCallback(
+                Control.fromValue(event.control), cancelledTriggers, null, request.extra
+            )
+        }
+
+        while (true) {
+            val receive = channel.receive()
+            responseObserver.onNext(StreamEventsResponse.newBuilder().setEvent(receive).build())
+            if (channel.isEmpty) {
+                break
+            }
+        }
+
+        cleanup()
+        responseObserver.onCompleted()
+    }
+
+    override fun getControls(
+        request: GetControlsRequest, responseObserver: StreamObserver<GetControlsResponse>
+    ) {
+        val controller = getResource(Controller.named(request.controller))
+        val controls = controller.getControls(request.extra)
+        responseObserver.onNext(GetControlsResponse.newBuilder().addAllControls(controls.map { it.value }).build())
+        responseObserver.onCompleted()
+    }
+
+    override fun getEvents(
+        request: GetEventsRequest, responseObserver: StreamObserver<GetEventsResponse>
+    ) {
+        val controller = getResource(Controller.named(request.controller))
+        val events = controller.getEvents(request.extra)
+        responseObserver.onNext(GetEventsResponse.newBuilder().addAllEvents(events.values.map { it.proto() }).build())
+        responseObserver.onCompleted()
+    }
+
+    override fun triggerEvent(
+        request: TriggerEventRequest, responseObserver: StreamObserver<TriggerEventResponse>
+    ) {
+        val controller = getResource(Controller.named(request.controller))
+        try {
+            controller.triggerEvent(Event.fromProto(request.event), request.extra)
+        } catch (e: Exception) {
+            throw e
+        }
+        responseObserver.onNext(TriggerEventResponse.newBuilder().build())
+        responseObserver.onCompleted()
+    }
+
+    override fun doCommand(
+        request: DoCommandRequest, responseObserver: StreamObserver<DoCommandResponse>
+    ) {
+        val controller = getResource(Controller.named(request.name))
+        val result = controller.doCommand(request.command.fieldsMap)
+        responseObserver.onNext(DoCommandResponse.newBuilder().setResult(result).build())
+        responseObserver.onCompleted()
+    }
+
+    override fun getGeometries(
+        request: GetGeometriesRequest, responseObserver: StreamObserver<GetGeometriesResponse>
+    ) {
+        val controller = getResource(Controller.named(request.name))
+        val result = controller.getGeometries(Optional.of(request.extra))
+        responseObserver.onNext(GetGeometriesResponse.newBuilder().addAllGeometries(result).build())
+        responseObserver.onCompleted()
+    }
+
+    override fun getResourceClass(): Class<Controller> {
+        return Controller::class.java
+    }
+
+    override fun getManager(): ResourceManager {
+        return this.manager
+    }
+}
```

Comment:
(minor) here and in subsequent files, we should add a newline character here for [POSIX compliance](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline).